crypto/asn1/a_gentm.c

```diff
@@ -22,7 +22,7 @@ int asn1_generalizedtime_to_tm(struct tm *tm, const ASN1_GENERALIZEDTIME *d)
     static const int min[9] = { 0, 0, 1, 1, 0, 0, 0, 0, 0 };
     static const int max[9] = { 99, 99, 12, 31, 23, 59, 59, 12, 59 };
```

Comment:
I get your point. It makes sense if the current code can mark date string like '20160231120000Z' as invalid and it might be fixed. But you would probably agree with me that we should not accept '20161231235960Z' just because the current code accepts '20160231120000Z'-like stuff. So I would like to leave it in this PR and might solve this later in other thread.